pkg/controllermanager/controller/project/stale/reconciler.go

```diff
@@ -151,9 +150,40 @@ func (r *Reconciler) projectInUseDueToBackupEntries(ctx context.Context, namespa
 	return kubernetesutils.ResourcesExist(ctx, r.Client, &gardencorev1beta1.BackupEntryList{}, r.Client.Scheme(), client.InNamespace(namespace))
 }
 
+func (r *Reconciler) projectInUseDueToWorkloadIdentities(ctx context.Context, namespace string) (bool, error) {
+	workloadIdentityList := &metav1.PartialObjectMetadataList{}
+	workloadIdentityList.SetGroupVersionKind(securityv1alpha1.SchemeGroupVersion.WithKind("WorkloadIdentityList"))
+	err := r.Client.List(
+		ctx,
+		workloadIdentityList,
+		client.InNamespace(namespace),
+		gardenerutils.UncontrolledSecretSelector,
+		client.MatchingLabels{v1beta1constants.LabelCredentialsBindingReference: "true"},
+	)
+	if err != nil {
+		return false, err
+	}
+
+	workloadIdentityNames := sets.New[string]()
+	for _, workloadIdentity := range workloadIdentityList.Items {
+		workloadIdentityNames.Insert(workloadIdentity.Name)
+	}
+	if workloadIdentityNames.Len() == 0 {
+		return false, nil
+	}
```

Comment:
Nit: This early exit could be before the `sets.Set` construction, right?

Reply:
Yes, you are right. Should be addressed by https://github.com/gardener/gardener/pull/11962/commits/89a427aa8269e7be119ddb0f111106d567b94ad3